File: problems/golang/11.binary-tree/main/find_level_order_test.go
package main

import (
	"github.com/AmirAhmadzadeh/problems/binary_tree"
	"reflect"
	"testing"
)

func TestFindLevelOrder(t *testing.T) {
	binaryTree := binary_tree.New(3)
	root := binaryTree.GetRoot()
	root.PushRightLeaf(1).PushRightLeaf(4)
	generatedNode := root.PushLeftLeaf(6)
	generatedNode.PushRightLeaf(2)
	generatedNode.PushLeftLeaf(9).PushRightLeaf(5).PushLeftLeaf(8)
	result := findLevelOrder(binaryTree)
	expectedResult := [][]int{{3}, {6, 1}, {9, 2, 4}, {5}, {8}}

	if !reflect.DeepEqual(expectedResult, result) {
		t.Errorf("Expected (%v) is not same as"+
			" actual string (%v)", expectedResult, result)
	}
}

File: problems/ts/11.binary-tree/findLevelOrder.test.ts
import { NodeImpl, Node, BinaryTreeImpl, findLevelOrderValues } from './';
import { test, assert } from 'vitest';

test('Find level order', () => {
    let binaryTree = new BinaryTreeImpl<number>(new NodeImpl(3));
    const root = binaryTree.getRoot() as Node<number>;
    root.pushRightLeaf(1).pushRightLeaf(4);
    const generatedNode = root.pushLeftLeaf(6);
    generatedNode.pushRightLeaf(2);
    generatedNode.pushLeftLeaf(9).pushRightLeaf(5).pushLeftLeaf(8);

    assert.deepEqual(findLevelOrderValues<number>({ ...root }), [
        [3],
        [6, 1],
        [9, 2, 4],
        [5],
        [8],
    ]);
});
